internal/provider/resource_gitlab_project_runner_enablement.go

```diff
@@ -0,0 +1,144 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_project_runner_enablement", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_project_runner_enablement`" + ` resource allows to manage the linking between a project and a runner instance.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/runners.html#enable-a-runner-in-project)`,
+		CreateContext: resourceGitlabProjectRunnerEnablementCreate,
+		ReadContext:   resourceGitlabProjectRunnerEnablementRead,
+		DeleteContext: resourceGitlabProjectRunnerEnablementDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The ID or URL-encoded path of the project owned by the authenticated user",
+			},
+			"runner_id": {
+				Type:        schema.TypeInt,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The ID of a runner to associate to the project",
+			},
+		},
+	}
+})
+
+func resourceGitlabProjectRunnerEnablementCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	projectID := d.Get("project").(string)
+	runnerID := d.Get("runner_id").(int)
+	options := &gitlab.EnableProjectRunnerOptions{
+		RunnerID: runnerID,
+	}
+
+	log.Printf("[DEBUG] create gitlab project runner %v/%v", projectID, runnerID)
+
+	_, _, err := client.Runners.EnableProjectRunner(projectID, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	runnerIDString := strconv.Itoa(runnerID)
+	d.SetId(buildTwoPartID(&projectID, &runnerIDString))
+
+	return resourceGitlabProjectRunnerEnablementRead(ctx, d, meta)
+}
+
+func resourceGitlabProjectRunnerEnablementRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	runnerType := "project_type"
+	projectID, runnerID, err := projectAndRunnerFromID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	options := &gitlab.ListProjectRunnersOptions{
+		ListOptions: gitlab.ListOptions{
+			PerPage: 20,
+			Page:    1,
+		},
+		Type: &runnerType,
+	}
+
+	log.Printf("[DEBUG] read gitlab project runner %s/%v", projectID, runnerID)
+
+	// We're enumerating all the runners for a project to make sure the runner
+	// is indeed associated with the project.
+
+	found := false
+	for {
```

Comment:
Can you refactor the `for`-loop to use the pattern used in other places, like here:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/f283205925feacd2bd553541ee2df50986a9106b/internal/provider/data_source_gitlab_project_issues.go#L350